src/components/Composer/index.tsx

```diff
@@ -345,13 +345,20 @@ function Composer(
                 disabled={isDisabled}
                 onKeyPress={handleKeyPress}
                 onFocus={(e) => {
-                    ReportActionComposeFocusManager.onComposerFocus(() => {
-                        if (!textInput.current) {
-                            return;
-                        }
-
-                        textInput.current.focus();
-                    });
+                    if (isReportActionCompose) {
+                        ReportActionComposeFocusManager.onComposerFocus(null);
+                    } else {
+                        // While a user was editing a comment and if they open on LHN menu we want the focus to return
+                        // to the ReportActionItemMessageEdit compose after they click on the menu (for e.g. mark as read)
+                        // so we assign the focus callback here.
+                        ReportActionComposeFocusManager.onComposerFocus(() => {
+                            if (!textInput.current) {
+                                return;
+                            }
+
+                            textInput.current.focus();
+                        });
+                    }
```

Comment:
```suggestion
                      ReportActionComposeFocusManager.onComposerFocus(() => {
                           /*
                             While a user was editing a comment and if they open on LHN menu we want the focus to return
                             to the ReportActionItemMessageEdit compose after they click on the menu (for e.g. mark as read)
                             so we assign the focus callback here.
                          */
                          if (!textInput.current || isReportActionCompose) {
                              return;
                          }

                          textInput.current.focus();
                      });
```

Reply:
Nope. What we are trying to avoid for main composer is overriding of the focus callback; `onComposerFocus` is callback assigner so we should avoid overriding of `mainComposerFocusCallback` by `focusCallback` so when `isReportActionCompose` is true we reset it to `null`.

Reply:
I still don't get the reasoning behind this. Why can't we just return early in the callback if `isReportActionCompose` is `true`?

Reply:
What we want to achieve is when the focus is called from reply in thread we want to focus the compose in the newly created thread and we can do that via `mainComposerFocusCallback` set here
https://github.com/Expensify/App/blob/822cf62de4e25558c7ffecb569d68c56e4e45d8e/src/pages/home/report/ReportActionCompose/ComposerWithSuggestions/ComposerWithSuggestions.tsx#L611-L617
But mainComposerFocusCallback are only called when `focusCallback` is not set.
https://github.com/Expensify/App/blob/822cf62de4e25558c7ffecb569d68c56e4e45d8e/src/libs/ReportActionComposeFocusManager.ts#L38-L43
So here what we want is to reset `focusCallback` to null so that our mainComposerFocusCallback will be called. But early returning, as U suggest, means we leave (without reseting) what was already set on `focusCallback` therefore our `mainComposerFocusCallback` will not be called.